core/src/main/resources/error/error-classes.json

```diff
@@ -789,7 +789,7 @@
   },
   "INVALID_PARAMETER_VALUE" : {
     "message" : [
-      "The value of parameter(s) '<parameter>' in <functionName> is invalid: <expected>"
+      "The value of parameter(s) '<parameter>' in <functionName> is invalid: <value>"
```

Comment:
I think we're trying to put a square peg into a round hole here.
I see three choices:
1) Add subclasses
2) Add dedicated error classes for these special case functions which speak to their unique requirements
3) a combo of 1) and 2)

Reply:
The four scenarios used in Spark are divided into four subclasses?
- aesCryptoError => INVALID_PARAMETER_VALUE.AES_KEY_BAD
- invalidPatternError => INVALID_PARAMETER_VALUE.INVALID_PATTERN
- invalidAesKeyLengthError => INVALID_PARAMETER_VALUE.AES_KEY_LENGTH_MISMATCH
- zeroArgumentIndexError => INVALID_PARAMETER_VALUE.ZERO_ARG_INDEX

Can I do that? @MaxGekk @srielau